### PR TITLE
Cleaned up _get_cache a bit.  Touch faster, smaller, readable.

### PR DESCRIPTION
**Pull request by thebeline**
Cleaned up _get_cache a bit. Touch faster, smaller, readable.

### CODE REVIEW

File: Minify_MinifiedFileRequestHandler.php [hunk 1]
```diff
@@ -620,98 +620,19 @@ function _get_minify_source( $file_path, $url ) {
...
-				$cache[0] = new \Minify_Cache_File(
+		static $cache = null;
+
+		if ( !isset( $cache ) ) {
+			$engine = $this->_config->get_string( 'minify.engine' );
+			
+			if ($engine == 'memcached') {
+				if ( !class_exists( 'Memcached' ) ) $engine = 'memcache';
+			} elseif ($engine == 'apc') {
+				if ( !function_exists( 'apc_store' ) ) $engine = 'apcu';
```
Comment (nigrosimone) [inline Minify_MinifiedFileRequestHandler.php:631]: what happen if user has APCu? `function_exists(` 'apc_store' )` return false.

Comment (nigrosimone) [reply Minify_MinifiedFileRequestHandler.php:631]: @thebeline I look forward to hearing from you.

Comment (thebeline) [reply Minify_MinifiedFileRequestHandler.php:631]: `!`, it will return false and assign `apcu`.

Comment (thebeline) [reply Minify_MinifiedFileRequestHandler.php:631]: If the test returns `false`, `!` makes it return `true`, which assigns `apcu`

**Comment by amiga-500** [comment]
Looks all right.  Thanks @thebeline ... nice job in the cleaning.  Very elegant.

**Pull request merged**